sdks/python/apache_beam/runners/common.py

```diff
@@ -218,10 +218,11 @@ def __init__(self, obj_to_invoke, method_name):
       elif core.DoFn.KeyParam == v:
         self.key_arg_name = kw
       elif isinstance(v, core.DoFn.RestrictionParam):
-        self.restriction_provider = v.restriction_provider
+        self.restriction_provider = v.restriction_provider or obj_to_invoke
```

Comment:
It might be less likely to happen but I'm curious what if the sdk author provides different `RestrictionProvider` from both ways, which one should be dominant. From implementation here, it seems like we will pick the param one.

Reply:
`v.restriction_provider` will be `None` iff `None` was passed to `DoFn.RestrictionParam` (or omitted, as `None` is the default value).

Reply:
I understand. The case I'm wondering is like:
```python
class MySDF(beam.DoFn, TypeARestrictionTracker):
  def process(restriction_tracker=RestrictionParam(TypeBRestrictionTracker()))
```

In this case, the sdk will pick up `TypeBRestrictionTracker` implementation. Is it ideal? I thought we want to pick up `TypeARestrictionTracker` in this case or just say the DoFn is not valid.